Framework/src/Check.cxx

```diff
@@ -88,12 +90,15 @@ void Check::initConfig(std::string checkName)
   mNumberOfTaskSources = 0;
   for (const auto& [_key, dataSource] : checkConfig.get_child("dataSource")) {
     (void)_key;
-    if (dataSource.get<std::string>("type") == "Task" || dataSource.get<std::string>("type") == "ExternalTask") {
+    if (auto sourceType = dataSource.get<std::string>("type");
+        sourceType == "Task" || sourceType == "ExternalTask" || sourceType == "PostProcessing") {
```

Comment:
I see what you mean with "it is not very elegant all these OR conditions" however I am glad to know that we can declare a variable inside a if clause, this is new to me.

Reply:
This is since c++17